### PR TITLE
API 659: Delete an unused App

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | OK
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Apps/back/tests/Acceptance/Context/AppContext.php [hunk 1]
```diff
@@ -117,9 +133,9 @@ public function iFindTheApp(string $label): void
     }
 
     /**
-     * @Then the App :label should exists
```
Comment (juliensnz) [inline src/Akeneo/Apps/back/tests/Acceptance/Context/AppContext.php]: Why that?

File: src/Akeneo/Apps/back/Infrastructure/Persistence/InMemory/Repository/InMemoryAppRepository.php [hunk 2]
```diff
@@ -57,6 +57,15 @@ public function update(App $app): void
         $this->dataRows[(string) $app->code()]['flow_type'] = (string) $app->flowType();
     }
 
+    public function delete(App $app): void
+    {
+        if (!isset($this->dataRows[(string) $app->code()])) {
+            throw new \LogicException(sprintf('App "%s" never persisted!', (string) $app->code()));
+        }
```
Comment (fitn) [inline src/Akeneo/Apps/back/Infrastructure/Persistence/InMemory/Repository/InMemoryAppRepository.php]: It should have the same behavior than the DbalAppRepository so if the row does not exist, we don't care ;)

Comment (pierrejolly) [reply src/Akeneo/Apps/back/Infrastructure/Persistence/InMemory/Repository/InMemoryAppRepository.php]: Indeed. Maybe I should change the "update" function  just above too?

File: src/Akeneo/Apps/back/tests/Unit/spec/Application/Command/DeleteAppHandlerSpec.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public function it_is_initializable(): void
+    {
+        $this->shouldHaveType(DeleteAppHandler::class);
+    }
+
+    public function it_deletes_an_app(
+        $repository,
+        $deleteClient,
+        $deleteUser,
+        App $magentoApp
```
Comment (pchasle) [inline src/Akeneo/Apps/back/tests/Unit/spec/Application/Command/DeleteAppHandlerSpec.php]: In general, we try to create models directly when possible and avoid using mock. It's more robust in the long run.

Comment (pierrejolly) [reply src/Akeneo/Apps/back/tests/Unit/spec/Application/Command/DeleteAppHandlerSpec.php]: Fixed

**Pull request merged**